frontend/javascripts/viewer/model/reducers/skeletontracing_reducer_helpers.ts

```diff
@@ -627,8 +628,10 @@ export function addTreesAndGroups(
       nodeId: idMap[bp.nodeId],
     }));
 
-    // Assign the new group id to the tree if the tree belongs to a group
-    tree.groupId = tree.groupId != null ? groupIdMap[tree.groupId] : tree.groupId;
+    // Assign the new group id to the tree if the tree belongs to a group that was newly added
+    // or keep the old group id if the tree should be assigned to an existing group.
+    tree.groupId =
+      tree.groupId != null && assignNewGroupId ? groupIdMap[tree.groupId] : tree.groupId;
```

Comment:
the else branch would be `a = a` which is superfluous. this variant is clearer imo:

```suggestion
    if (tree.groupId != null && assignNewGroupId) {
      tree.groupId = groupIdMap[tree.groupId];
    }
```